src/components/input/button.module.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { PropsWithChildren } from "react";
+import styles from "./button.module.scss";
+
+
+const Button: React.FC<PropsWithChildren<Props>> = ({
+  children,
+  size = "normal",
+  variant = "primary",
+  type = "button",
+  onClick = () => undefined,
+}) => {
+
+  return (
+    <button
+      className={styles.main}
+
+      type={type}
+      data-variant={variant}
+      data-size={size}
+
+      onClick={onClick}
+    >
+      {children}
+    </button>
+  );
+};
+
+export { Button };
+
+interface Props {
+  variant?: "primary" | "secondary";
+  type?: "button" | "submit" | "reset";
+  size?: "small" | "normal" | "big";
```

Comment:
`size` should be extracted to a shared type of we're gonna use it in multiple places.

```ts
export type ElementSize = "small" | "normal" | "big";
```

It may also be nicer to use if they were named `sm`, `md`, and `lg`. That way it'll remain consistent if we ever were to add smaller or larger sizes (xs vs tiny, xl vs huge?). This is just a suggestion though, based on what I've seen in other component libraries / style guides